### ISSUE TITLE
Яндекс.Практикум:Код-ревью.

### ISSUE THREAD

**Issue by zlocate** [AngelinaBrichko/10project#1]
Добрый день :)
Пожалуйста, ведите работу над проектом в одном репозитории. (предыдущий вариант был выполнен в project10). Несоблюдение этого правила приводит к тому, что мои комментарии теряются и оценивать прогресс становится сложнее.
Спасибо!

Если у вас возникли сложности при работе с Git - дайте знать, постараюсь помочь :)

Предыдущие комментарии:
Здравствуйте. Отличная работа! :)
**Отлично** Все ветки присутствуют в репозитории.
**Отлично** Все регулярные выражения не имеют грубых ошибок, которые делают их невалидными.

К сожалению текущая реализация содержит ряд недостатков, не позволяющих мне засчитать работу на данной итерации:

Репозиторий:
**Нужно исправить** все три ветки должны содержать одинаковую ревизию проекта.
Работа выполняется в ветке develop, после чего изменения сливаются в master и gh-pages.
Спасибо
**Нужно исправить:** отсутствуют git-теги отражающие номер версии. Создать их можно при помощи команды `git tag -a v0.0.1 -m "my first version"`
Обращаю внимание что теги по-умолчанию не публикуются при вызове команды `git push`
Для их публикации выполните команду `git push origin --tags`
Подробнее: https://git-scm.com/book/ru/v2/%D0%9E%D1%81%D0%BD%D0%BE%D0%B2%D1%8B-Git-%D0%A0%D0%B0%D0%B1%D0%BE%D1%82%D0%B0-%D1%81-%D0%BC%D0%B5%D1%82%D0%BA%D0%B0%D0%BC%D0%B8

 **Нужно исправить**: README файл должен содержать номер версии (ненулевой), ссылку на проект (github pages) и **заголовок отражающий назначение проекта**.

**Можно лучше** файлы logo.svg и cover.jpg, расположенные в репозитории не используются на страничке (загружаются с серверов Яндекса). Просьба либо задействовать их либо удалить

**Можно лучше**: .gitignore должен содержать правила, исключающие попадание временных и\или системных файлов в репозиторий. 

Регулярные выражения:
**Нужно исправить**: каждое регулярное выражение должно начинаться со знака ^ и заканчиваться знаком $, чтобы предотвратить частичное совпадение вводимого значения и пропускать ввод только при полном совпадении с регулярным выражением.
**Можно лучше** Квантификатор {0,} можно заменить на *

Имя:
**Нужно исправить** Валидацию не проходят следующие корректные имена:
Ёлка-Ёлка

Email:
**Нужно исправить** Валидацию не проходят следующие корректные адреса
some@ya-2.ru - доменное имя может содержать цифры и `-`. Однако не может состоять только из `-`, а также начинаться и заканчиваться на `-`.
Т.е some@a-b.ru корректно, а some@-a-b-.ru - нет

**Нужно исправить** Валидацию проходят следующие некорректные адреса
-_-@x.ru - имя не может состоять только из спецсимволов, а также начинаться и заканчиваться на них

Номер телефона:
**Нужно исправить** Данные форматы номеров телефонов проходят валидацию, хотя с точки зрения постановки задачи они являются невалидными:
+7(925 900-90-90
+7925900-90-90
+7 (925) 900-90-90

Адрес сайта:
**Нужно исправить** Валидацию не проходят следующие корректные адреса
http://google.ru/ - адрес со слешем на конце является корректным (не допускается несколько слешей подряд)
http://some.test.yandex.ru - адреса сайтов с поддоменами являются корректными.
https://x.academy:20000/something - путь не обязательно должен заканчиваться на символ #

**Comment by zlocate** [comment]
**Нужно исправить** валидацию не проходят следующие корректные номера:
+7(925) 900-90-90 

Почта:
**Нужно исправить**: регулярное выражение содержит синтаксические ошибки, из-за чего валидация сломана
Текст ошибки:
```
Pattern attribute value ^[a-z]([\w-.]+)[a-z0-9]@[a-z0-9-]+(\.[a-z]+)$ is not a valid regular expression: Uncaught SyntaxError: Invalid regular expression: /^[a-z]([\w-.]+)[a-z0-9]@[a-z0-9-]+(\.[a-z]+)$/: Invalid character class
```
Скриншот консоли https://prnt.sc/rvvvk6

После исправления просьба дополнительно проверить данные кейсы:
`_@site.ru` - имя ящика не может состоять только из спецсимволов
`_some_@ya.ru` - имя ящика не может начинаться и заканчиваться на спецсимвол
`mymail@сайт.ru` - разрешены только латинские буквы. ввод кириллицы недопустим в рамках задачи.
`my---mail@site.ru` - адрес электронной почы, содержащий несколько символов '-' подряд является некорректным
`some_address@mail` - отсутствует доменная зона (например .ru/.com и т.п)


Адрес сайта:
**Нужно исправить**: регулярное выражение содержит синтаксические ошибки, из-за чего валидация сломана
Текст ошибки:
```
Pattern attribute value ^(http(s)?:\/\/)(www\.)?((([\w-\.]+\.[a-z]+)|((\d{1,3}\.){3}\d{1,3}))(:[^0]\d{2,5})?([\w]/[^_]{0,}#?)?[\w/#]?)$ is not a valid regular expression: Uncaught SyntaxError: Invalid regular expression: /^(http(s)?:\/\/)(www\.)?((([\w-\.]+\.[a-z]+)|((\d{1,3}\.){3}\d{1,3}))(:[^0]\d{2,5})?([\w]/[^_]{0,}#?)?[\w/#]?)$/: Invalid character class
```
Скриншот консоли:
https://prnt.sc/rvvvk6

Самопроверка: открыть консоль браузера, переключиться в ней на вкладку Console, ввести в проверяемое поле любой текст. Если в консоли появилась ошибка - регулярное выражение содержит синтаксическую ошибку

**Comment by AngelinaBrichko** [comment]
@zlocate Добрый день. Продолжить в этом репозитории?

**Comment by zlocate** [comment]
@AngelinaBrichko да, можете продолжить в этом. Если не секрет, почему решили в новый репозиторий переехать. Если были какие-то сложности лучше разобраться сейчас, чтобы в дальнейшем чувствовать себя комфортно.

**Comment by AngelinaBrichko** [comment]
@zlocate да, были сложности(; сейчас попробую разобраться

**Comment by AngelinaBrichko** [comment]
@zlocate Я ничего еще в коде не исправляла, но у меня нет ошибок в консоли, либо делаю что-то не так https://prnt.sc/rvxjp8 (пробовала и в safari и в google)

**Comment by zlocate** [comment]
@AngelinaBrichko проблема проявлялась для двух полей в актуальном Firefox
https://prnt.sc/rxh6b7

После ваших правок проблема осталась только для поля адрес сайта:
https://prnt.sc/rxh85m
Просьба исправить и  вернуть на проверку.
З.Ы Извините за поздний ответ, очень много уведомлений с гитхаба падает на почту.
Для того чтобы оперативно получить ответ можете заливать работу повторно (я не буду в таком случае считать итерации :) )

**Comment by zlocate** [comment]
**Можно лучше** побилась верстка заголовка в README. 
Для того чтобы избежать подобных моментов рекомендую проверять результат перед изменением (если делаете в веб-интерфейсе можно проверить на вкладке Preview, если нет - в визуальном редакторе StackEdit (https://stackedit.io/))

**Comment by zlocate** [comment]
**Можно лучше** в репозиторий попал временный файл .DS_Store. Он не несет полезной нагрузки, поэтому рекомендую его удалить и добавить соответствующее правило в .gitignore

**Comment by zlocate** [comment]
Отлично! Проблемы в Firefox решены.
Остались следующие кейсы:
Email:
**Нужно исправить** валидацию не проходят следующие корректные адреса
some@some.ya.ru - домменнное имя в адресе почты может содержать поддомены

**Можно лучше** валидацию проходят некорректные адреса сайтов содержащие несколько слешей или # на конце
http://ya.ru:98/////////////////######## 

**Нужно исправить** валидацию не проходят следующие корректные адреса сайтов:
http://127.0.0.1

**Нужно исправить** валидацию проходят следующие некорректные адреса сайта:
`http://www._.ru:000000000000000` - некорректное доменное имя, некорректный номер порта
`https://_.s_______________:00/` некорректная доменная зона (доменная зона не может содержать спецсимволов)

После их исправления работа будет принята

**Comment by zlocate** [comment]
Добрый день. Недостатки устранены. Работа принята!
Желаю вам успехов в дальнейшем обучении. :)
